### PR TITLE
patch init.d scripts to support ubuntu 15.10

### PR DESCRIPTION
**Pull request by wiwiwa**
Fix: install script not working on ubuntu 15.10
Reason:
- /sbin/start-stop-daemon interprets console redirection operator '<' as arguments to service executable. 
- several service not automatically started by script

### CODE REVIEW

File: cluster/ubuntu/util.sh [hunk 1]
```diff
@@ -306,10 +306,7 @@ function kube-up() {
   source "${KUBE_ROOT}/cluster/ubuntu/${KUBE_CONFIG_FILE:-config-default.sh}"
 
   # downloading tarball release
-  if [[ -d "${KUBE_ROOT}/cluster/ubuntu/binaries" ]]; then
-    rm -rf "${KUBE_ROOT}/cluster/ubuntu/binaries"
-  fi
-  "${KUBE_ROOT}/cluster/ubuntu/download-release.sh"
```
Comment (dalanlan) [inline cluster/ubuntu/util.sh]: This is intentional. We don't want to use the out-dated binaries when the user reset $KUBE_VERSION var, and it won't easy to trace the corresponding version of the binaries. Too much chaos.

File: cluster/ubuntu/util.sh [hunk 2]
```diff
@@ -306,10 +306,7 @@ function kube-up() {
...
+  [[ -d "${KUBE_ROOT}/cluster/ubuntu/binaries" ]] || "${KUBE_ROOT}/cluster/ubuntu/download-release.sh"
```
Comment (dalanlan) [inline cluster/ubuntu/util.sh]: Drop this, please.

File: cluster/ubuntu/util.sh [hunk 3]
```diff
@@ -508,6 +505,12 @@ function provision-masterandnode() {
       cp ~/kube/minion/* /opt/bin/
 
       service etcd start
+      service kube-apiserver start
```
Comment (dalanlan) [inline cluster/ubuntu/util.sh]: Don't bother bring up the k8s components one by one. They're triggered by `etcd`.

**Comment by wiwiwa** [comment]
I signed it!

File: cluster/ubuntu/util.sh (hunk 3 above)
Comment (wiwiwa) [inline cluster/ubuntu/util.sh]: I think this script is just a script for newbies. Newbies can not set up a cluster without these lines, according to the offical document. They, like me, are just in trouble. 

What does 'triggered by etcd' mean? Will `etcd` start them automatically?

File: cluster/ubuntu/util.sh (hunk 2 above)
Comment (wiwiwa) [inline cluster/ubuntu/util.sh]: How can I drop these code? Do I need create another pull request?

Shall I change the code to skip download if not modified? We, in China, need a over 15 minutes to download the file. If something wrong when running `kube-up.sh`, we have try again and again. So it not acceptable to always download the file.

File: cluster/ubuntu/util.sh (hunk 3 above)
Comment (dalanlan) [inline cluster/ubuntu/util.sh]: You won't suffer. The script takes care of this for you.
Ref https://github.com/kubernetes/kubernetes/blob/master/cluster/ubuntu/master/init_conf/kube-apiserver.conf#L7

File: cluster/ubuntu/util.sh (hunk 2 above)
Comment (dalanlan) [inline cluster/ubuntu/util.sh]: Your code is *not* correct. If you insist, you should handle cross validation between local binaries and $KUBE_VERSION
Moreover you should take care of the networking issue yourself.

File: cluster/ubuntu/util.sh (hunk 3 above)
Comment (wiwiwa) [inline cluster/ubuntu/util.sh]: OK, great idea. But it does not work on ubuntu 15.10. I'd like to write some systemd unit files.
I will close this pull requests and create a new pull request on finish.

Thanks

**Comment by wiwiwa** [comment]
As ubuntu 15.10 does not support upstart scripts, I will create some systemd unit files.

File: cluster/ubuntu/util.sh (hunk 3 above)
Comment (dalanlan) [inline cluster/ubuntu/util.sh]: We make it clear in document about this. I had no experience with ubuntu 15.10 though.
